asv/benchmark.py

```diff
@@ -704,16 +707,64 @@ def __init__(self, name, func, attr_sources):
         self.unit = "bytes"
 
     def run(self, *param):
-        # We can't import asizeof directly, because we haven't loaded
-        # the asv package in the benchmarking process.
-        path = os.path.join(
-            os.path.dirname(__file__), 'extern', 'asizeof.py')
-        asizeof = importlib.machinery.SourceFileLoader('asizeof', path).load_module()
+        if ON_PYPY:
+            raise NotImplementedError("asizeof doesn't work on pypy")
+            return
+
+        def import_asizeof():
+            """Import asizeof, searching system Pythons in PATH."""
```

Comment:
Why do we need all this? If the import fails, can't we just fail the run?

Reply:
The problem here is that the import fails where it shouldn't, e.g. on the CI. This seems to be because of the process isolation (each ASV run is in a different subprocess). Perhaps, now that I'm thinking about it, it would be cleaner to instead import `asv` itself, which would automatically bring in `pympler` and is discussed in https://github.com/airspeed-velocity/asv/issues/908

Reply:
If I went with that approach then the Import Mechanism would instead load ASV rather than `asizeof`, but I'm not sure how to simplify this further without breaking the subprocess approach.